src/PluginClient.php

```diff
@@ -79,7 +78,7 @@ public function sendRequest(RequestInterface $request)
         $pluginChain = $this->createPluginChain($this->plugins, function (RequestInterface $request) {
             try {
                 return new FulfilledPromise($this->client->sendRequest($request));
-            } catch (HttplugException $exception) {
+            } catch (\Exception $exception) {
```

Comment:
the promise interface would mean that HTTPlug starts to depend on promise, to implement that interface on its exceptions. what is the state of promise PSR?

i think we should not handle domain exception in HTTPlug plugins. the plugin chain is for handling HTTP, not for handling domain exceptions. the github api exceptions don't transport the response object - which is correct, but makes it really pointless to continue handling the exception in http plugins. PluginClient should not be an arbitrary stack of promises. if a user wants to build a stack of domain promises, they should not use http promises for that, imho.